pkg/leclient/lets_encrypt.go

```diff
@@ -0,0 +1,199 @@
+		contacts = append(contacts, "mailto:"+email)
+	}
+
+	c.Account, err = c.Client.UpdateAccount(c.Account, true, contacts...)
+	return err
+}
+
+func (c *ACMEClient) CreateOrder(domains []string) (err error) {
+	var certDomains []string
+	var ids []acme.Identifier
+
+	for _, domain := range domains {
+		certDomains = append(certDomains, domain)
+		ids = append(ids, acme.Identifier{Type: "dns", Value: domain})
+	}
+	c.Order, err = c.Client.NewOrder(c.Account, ids)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *ACMEClient) GetAccount(kubeClient client.Client, staging bool, namespace string) (err error) {
+	accountURL, err := getLetsEncryptAccountURL(kubeClient, true)
+	if err != nil {
+		return err
+	}
+
+	privateKey, err := getLetsEncryptAccountPrivateKey(kubeClient, true)
+	if err != nil {
+		return err
+	}
+	c.Account = acme.Account{PrivateKey: privateKey, URL: accountURL}
+	return nil
+}
+
+func (c *ACMEClient) GetOrderURL() (URL string, err error) {
+	URL = c.Order.URL
+	return URL, err
+}
+
+func (c *ACMEClient) OrderAuthorization() []string {
+	return c.Order.Authorizations
+}
+
+func (c *ACMEClient) FetchAuthorization(authURL string) (err error) {
+	c.Authorization, err = c.Client.FetchAuthorization(c.Account, authURL)
+	return err
+}
+func (c *ACMEClient) GetAuthorizationURL() string {
+	return c.Authorization.URL
+}
+func (c *ACMEClient) GetAuthorizationIndentifier() string {
+	return c.Authorization.Identifier.Value
+}
+func (c *ACMEClient) SetChallengeType() {
+	c.Challenge = c.Authorization.ChallengeMap["dns-01"]
+}
+func (c *ACMEClient) GetDNS01KeyAuthorization() string {
+	return acme.EncodeDNS01KeyAuthorization(c.Challenge.KeyAuthorization)
+}
+func (c *ACMEClient) GetChallengeURL() string {
+	return c.Challenge.URL
+}
+func (c *ACMEClient) UpdateChallenge() (err error) {
+	c.Challenge, err = c.Client.UpdateChallenge(c.Account, c.Challenge)
+	return err
+}
+func (c *ACMEClient) FinalizeOrder(csr *x509.CertificateRequest) (err error) {
+	c.Order, err = c.Client.FinalizeOrder(c.Account, c.Order, csr)
+	return err
+}
+func (c *ACMEClient) GetOrderEndpoint() string {
+	return c.Order.Certificate
+}
+func (c *ACMEClient) FetchCertificates() (certbundle []*x509.Certificate, err error) {
+	certbundle, err = c.Client.FetchCertificates(c.Account, c.Order.Certificate)
+	return certbundle, err
+}
+func (c *ACMEClient) RevokeCertificate(certificate *x509.Certificate) (err error) {
+	err = c.Client.RevokeCertificate(c.Account, certificate, c.Account.PrivateKey, 0)
+	return err
+}
+func GetLetsEncryptClient(staging bool) (Client ACMEClient, err error) {
+	if staging {
+		Client.Client, err = acme.NewClient(acme.LetsEncryptStaging)
+		return Client, err
+	}
+	Client.Client, err = acme.NewClient(acme.LetsEncryptProduction)
+	return Client, err
+}
+
+func getLetsEncryptAccountPrivateKey(kubeClient client.Client, staging bool) (privateKey crypto.Signer, err error) {
+
+	secretName := LetsEncryptProductionAccountSecretName
```

Comment:
Question: Do we expect an install of this operator to have both staging and prod? Is there a reason to have two different secrets in the same NS?

Reply:
No, it will never do both, which is why it needs to be streamlined out of the code. That's coming in another story/PR.